components/org.wso2.carbon.datasource.core/src/main/java/org/wso2/carbon/datasource/core/beans/JNDIConfig.java

```diff
@@ -115,23 +112,25 @@ public boolean equals(Object rhs) {
 
     @Override
     public int hashCode() {
-        assert false : "hashCode() not implemented";
+//        assert false : "hashCode() not implemented";
```

Comment:
As mentioned in the previous comment, is there a specific reason for returning -1. Furthermore having commented code is considered as a code smell and therefore IMO we should remove this line. WDYT?